window/src/os/x11/window.rs

```diff
@@ -1525,16 +1525,16 @@ impl XWindowInner {
             });
     }
 
-    fn set_resize_increments(&mut self, x: u16, y: u16) -> anyhow::Result<()> {
+    fn set_resize_increments(&mut self, x: u16, y: u16, base_width: u16, base_height: u16) -> anyhow::Result<()> {
         use xcb_util::*;
         let hints = xcb_size_hints_t {
-            flags: XCB_ICCCM_SIZE_HINT_P_RESIZE_INC,
+            flags: XCB_ICCCM_SIZE_HINT_P_MIN_SIZE | XCB_ICCCM_SIZE_HINT_P_RESIZE_INC | XCB_ICCCM_SIZE_HINT_BASE_SIZE,
             x: 0,
             y: 0,
             width: 0,
             height: 0,
-            min_width: 0,
-            min_height: 0,
+            min_width: (base_width + x).into(),
```

Comment:
FWIW, the macOS equivalent of this appears to be: https://developer.apple.com/documentation/appkit/nswindow/1419670-contentminsize?language=objc